src/monsters.cpp

```diff
@@ -323,31 +323,44 @@ bool Monsters::deserializeSpell(const pugi::xml_node& node, spellBlock_t& sb, co
 			combat->setParam(COMBAT_PARAM_TYPE, COMBAT_HEALING);
 			combat->setParam(COMBAT_PARAM_AGGRESSIVE, 0);
 		} else if (tmpName == "speed") {
-			int32_t speedChange = 0;
+			int32_t minSpeedChange = 0;
+			int32_t maxSpeedChange = 0;
 			int32_t duration = 10000;
 
 			if ((attr = node.attribute("duration"))) {
 				duration = pugi::cast<int32_t>(attr.value());
 			}
 
 			if ((attr = node.attribute("speedchange"))) {
-				speedChange = pugi::cast<int32_t>(attr.value());
-				if (speedChange < -1000) {
-					//cant be slower than 100%
-					speedChange = -1000;
+				minSpeedChange = maxSpeedChange = std::max(-1000, pugi::cast<int32_t>(attr.value()));
+			} else if ((attr = node.attribute("minspeedchange"))) {
+				minSpeedChange = maxSpeedChange = std::max(-1000, pugi::cast<int32_t>(attr.value()));
+
+				if ((attr = node.attribute("maxspeedchange"))) {
+					maxSpeedChange = std::max(-1000, pugi::cast<int32_t>(attr.value()));
+				}
+
+				if ((minSpeedChange ^ maxSpeedChange) < 0) {
+					std::cout << "[Error - Monsters::deserializeSpell] - " << description << " - invalid speed change range: " << name << std::endl;
```

Comment:
It's not printing the range. Also I think it's useful to print a warning above stating that the minimum change is -1000.

Reply:
Made initial changes to the error message:
```c++
std::cout << "[Error - Monsters::deserializeSpell] - " << description << " - invalid speed change range: minspeedchange=\"" << minSpeedChange << "\" maxspeedchange=\"" << maxSpeedChange << "\"" << std::endl;
```
Going to have to sit on this for a second to find an appropriate way to do the warning, and print the actual minSpeedChange instead of the value after the max function has been applied to it.